### ISSUE TITLE
Template binding bypasses element class get/set methods

### ISSUE THREAD

**Issue by felixc** [dart-lang/polymer-dart#67]
Dart supports class `get` and `set` methods for customizing access to object properties. Polymer support template binding to tie object properties to the DOM. These two features should interact nicely, but instead Polymer's template binding appears to bypass Dart's property methods.

In particular, setting an object attribute via a Polymer template binding doesn't invoke the class's `set` method.

A minimal example:
```dart
@CustomTag('my-example-element')
class MyExampleElement extends PolymerElement {

    @published
    String get myExampleProperty => readValue(#myExampleProperty);

    set myExampleProperty(String newValue) {
        writeValue(#myExampleProperty, newValue);
        print('Property value changed!');
    }
}
```

Now, if we do:
```dart
$['id-of-an-instance-of-my-example-element'].myExampleProperty = 'foo';
```
Our setter gets called, the value of the property is updated in any template bindings, and the `print()` message in the setter shows up.

However, if instead we do:
```
<my-example-element myExampleProperty="foo"></my-example-element>
```
Then our setter is *not* called, and although the value of the property is updated in template bindings and in code within the Dart class that makes use of it, the `print()` in the setter isn't invoked.

Ideally, these two should be equivalent, and they should both go through the class's `set` method.

**Comment by sigmundch** [comment]
Thanks for opening this issue! Just to bring here some of the information we discussed by email:

The reason behind this is that when you create a binding in the HTML, we create a binding object. This binding object stores the current value represented by the element attribute (the template-binding system assigns values directly to it). The get/set pair pattern with readValue/writeValue are not the source of truth for the value of the binding. They instead connect to the binding object to retrieve the value when you access the property programatically.

With all the changes we saw in polymer.js 0.8, we have room for making bigger changes in the future. We should consider new design options in this space for polymer-dart that would make this less convoluted.
